src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/DataSetServiceTests.cs

```diff
@@ -0,0 +1,205 @@
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.ReleaseFiles.AddRangeAsync(releaseFile);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    releaseFile.ReleaseId, releaseFile.FileId);
+                var viewModel = result.AssertRight();
+
+                Assert.Equal(releaseFile.Name, viewModel.Title);
+                Assert.Equal(releaseFile.Summary, viewModel.Summary);
+
+                var file = releaseFile.File;
+
+                Assert.Equal(file.Id, viewModel.File.Id);
+                Assert.Equal(file.Filename, viewModel.File.Name);
+                Assert.Equal(file.DisplaySize(), viewModel.File.Size);
+
+                Assert.Equal(releaseFile.ReleaseId, viewModel.Release.Id);
+                Assert.Equal(releaseFile.Release.Title, viewModel.Release.Title);
+                Assert.Equal(releaseFile.Release.Slug, viewModel.Release.Slug);
+                Assert.Equal(releaseFile.Release.Type, viewModel.Release.Type);
+                Assert.True(viewModel.Release.IsLatestPublishedRelease);
+                Assert.Equal(releaseFile.Release.Published, viewModel.Release.Published);
+
+                Assert.Equal(publication.Id, viewModel.Release.Publication.Id);
+                Assert.Equal(publication.Title, viewModel.Release.Publication.Title);
+                Assert.Equal(publication.Slug, viewModel.Release.Publication.Slug);
+                Assert.Equal(publication.Topic.Theme.Title, viewModel.Release.Publication.ThemeTitle);
+            }
+        }
+
+        [Fact]
+        public async Task NoRelease()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Publications.AddRangeAsync(publication);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    Guid.NewGuid(), Guid.NewGuid());
+
+                result.AssertNotFound();
+
+            }
+        }
```

Comment:
This could be testing no release _or_ no files.

I would still save a release here with a file and for the method under test make the following call.

```csharp
GetDataSet(releaseId: Guid.NewGuid(),
    fileId: releaseFile.FileId)
```